cpp/include/raft/core/cusparse_macros.hpp

```diff
@@ -35,6 +35,7 @@
 // (i.e., before including this header)
 //
 #define CUDA_VER_10_1_UP (CUDART_VERSION >= 10100)
+#define CUDA_VER_12_4_UP (CUDART_VERSION >= 12040)
```

Comment:
Shouldn't this be like this? 
```suggestion
#define CUDA_VER_12_4_UP (CUDART_VERSION >= 12400)
```

Reply:
Woops, sorry you're right, good catch.

I had tested with CUDA 12.0 which incorrectly showed that it was working. Should have tried with CUDA 12.2 or other

Reply:
https://nvidia.github.io/cuda-python/module/cudart.html#cuda.cudart.cudaRuntimeGetVersion
> The version is returned as (1000 * major + 10 * minor). For example, CUDA 9.2 would be represented by 9020

So for CUDA 12.4, `CUDART_VERSION = 12040`.
The line just above should be corrected to: `#define CUDA_VER_10_1_UP (CUDART_VERSION >= 10010)` (as well as line 62 of this same file)

Reply:
@lowener Ahaha you are right. I had written 12040 then forgot where I had found it to then corrected again to 12400 after the above comment.

Let me fix all those lines